src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -303,6 +309,19 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
    */
   detachMediaSource(): void {}
 
+  _setSrc(): void {
+    const pkRequest: PKRequestObject = {url: this._sourceObj ? this._sourceObj.url : '', body: null, headers: {}};
```

Comment:
the {url: url, body: body, headers: headers} object creation is very repetitive.
Create a static helper that returns the object and use it around.